system/HTTP/SiteURIFactory.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\HTTP;
+
+use CodeIgniter\HTTP\Exceptions\HTTPException;
+use CodeIgniter\Superglobals;
+use Config\App;
+
+class SiteURIFactory
```

Comment:
Please explain your scope choices - not making the class `final` but making most of its methods `private` instead of `protected` or `final protected`.

Reply:
I made it final.